### PR TITLE
DEV-14860: Upgrade provider versions

### PR DESCRIPTION
**Pull request by schwartzmanb**

### CODE REVIEW

File: examples/minimal/providers.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 provider "azurerm" {
-  version = "=2.11.0"
+  version = "=2.60.0"
   features {}
```
Comment (souza-dan) [inline examples/minimal/providers.tf:3]: Not related, but we should probably include our partner id in the examples.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#partner_id

File: modules/azure-service-principal/main.tf [hunk 2]
```diff
@@ -1,8 +1,6 @@
 resource "azuread_application" "adls-gen2-application" {
-  name                       = var.application_name
-  available_to_other_tenants = false
-  oauth2_allow_implicit_flow = true
-  type                       = "native"
+  display_name     = var.application_name
+  sign_in_audience = "AzureADMyOrg"
```
Comment (souza-dan) [inline modules/azure-service-principal/main.tf:3]: Any use case for setting this to `AzureADMultipleOrgs`?

**Review by souza-dan — APPROVED** [review]
LGTM

**Review by souza-dan — APPROVED** [review]

File: modules/azure-service-principal/main.tf (hunk 2 above)
Comment (schwartzmanb) [reply modules/azure-service-principal/main.tf:3]: I don't think so, I don't see a reason to allow access to tenants outside of the org we're deploying in

**Pull request merged**